src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,1191 @@
+
+        # STEP 7: TH reads from the DUT the CountdownTime attribute
+        self.step(7)
+        if self.pics_guard(self.check_pics("%s.S.A0002" % self.test_info.pics_code)):
+            initial_countdown_time = await self.read_expect_success(endpoint=endpoint,
+                                                                    attribute=attributes.CountdownTime)
+            if initial_countdown_time is not NullValue:
+                asserts.assert_true(0 <= initial_countdown_time <= 259200,
+                                    "CountdownTime(%s) must be between 0 and 259200" % initial_countdown_time)
+
+        # STEP 8: TH reads from the DUT the PhaseList attribute
+        self.step(8)
+        if self.pics_guard(self.check_pics("%s.S.A0000" % self.test_info.pics_code)):
+            phase_list = await self.read_expect_success(endpoint=endpoint,
+                                                        attribute=attributes.PhaseList)
+            phase_list_len = 0
+            if phase_list is not NullValue:
+                phase_list_len = len(phase_list)
+                asserts.assert_less_equal(phase_list_len, 32,
+                                          "PhaseList length(%d) must be less than 32!" % phase_list_len)
+
+        # STEP 9: TH reads from the DUT the CurrentPhase attribute
+        self.step(9)
+        if self.pics_guard(self.check_pics("%s.S.A0001" % self.test_info.pics_code)):
+            current_phase = await self.read_expect_success(endpoint=endpoint,
+                                                           attribute=attributes.CurrentPhase)
+            if (phase_list == NullValue) or (not phase_list):
+                asserts.assert_equal(current_phase, NullValue, "CurrentPhase(%s) should be null" % current_phase)
+            else:
+                asserts.assert_less_equal(0, current_phase,
+                                          "CurrentPhase(%s) must be greater or equal than 0" % current_phase)
+                asserts.assert_less(current_phase < phase_list_len,
+                                    "CurrentPhase(%s) must be less then %s" % (current_phase, (phase_list_len - 1)))
+
+        # STEP 10: TH waits for {PIXIT.WAITTIME.COUNTDOWN}
+        self.step(10)
+        if self.pics_guard(self.check_pics("%s.S.A0002" % self.test_info.pics_code)):
+            time.sleep(wait_time)
+
+        # STEP 11: TH reads from the DUT the CountdownTime attribute
+        self.step(11)
+        if self.pics_guard(self.check_pics("%s.S.A0002" % self.test_info.pics_code)):
+            countdown_time = await self.read_expect_success(endpoint=endpoint,
+                                                            attribute=attributes.CountdownTime)
+
+            if (countdown_time is not NullValue) and (initial_countdown_time is not NullValue):
+                difference = abs(countdown_time - initial_countdown_time)
```

Comment:
The `abs` will cause this test to pass if the time INCREASED instead of decreased. Please remove the abs.

Suggest `asserts.assert_less_equal(countdown_time, (initial_countdown_time - wait_time), f"The countdown time shall have decreased at least {wait_time:.1f} since start command")`

Reply:
done